lib/type.ex

```diff
@@ -0,0 +1,11 @@
+defprotocol Litmus.Type do
+  @moduledoc false
+
+  alias Litmus.Type
+
+  @type t :: %Type.Any{} | %Type.Boolean{} | %Type.Number{} | %Type.String{}
+
+  @doc "validates the data based on the type"
```

Comment:
I think we can leave out this doc for now. i'm not sure its relevant for end-users.

Reply:
@moduledoc is false over there. The end users will not be able to read it I think. I just kept it for the code documentation. Should I delete it?

Reply:
I would say so